internal/impl/babysitter.go

```diff
@@ -148,11 +151,19 @@ func (b *babysitter) Serve() error {
 }
 
 // ActivateComponent implements the envelope.EnvelopeHandler interface.
-func (b *babysitter) ActivateComponent(_ context.Context, request *protos.ActivateComponentRequest) (*protos.ActivateComponentReply, error) {
+func (b *babysitter) ActivateComponent(ctx context.Context, request *protos.ActivateComponentRequest) (*protos.ActivateComponentReply, error) {
 	go func() {
-		if err := b.watchPods(b.ctx, request.Component); err != nil {
-			// TODO(mwhittaker): Log this error.
-			fmt.Fprintf(os.Stderr, "watchPods(%q): %v", request.Component, err)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-time.After(5 * time.Second):
+				if err := b.watchPods(b.ctx, request.Component); err != nil {
+					// TODO(mwhittaker): Log this error.
+					fmt.Fprintf(os.Stderr, "watchPods(%q): %v", request.Component, err)
+					b.logger.Error("error watching pods", "err", err, "component", request.Component)
+				}
+			}
```

Comment:
I'm proposing here a retry on every 5 seconds, so, if the watchPods fails, we try again.
For me, or we do this, or we panic... allowing an error here to end the goroutine is giving the system a chance to hang forever again due to unexpected errors.

Reply:
sorry, it didn't work...